File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.cpp
// Last updated: 2/21/2026, 9:44:07 AM
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        
        int n=nums.size();
        int prefix[2][n];
        prefix[0][0]=1;
        prefix[1][n-1]=1;
        for(int i=1;i<n;i++)
        {
            prefix[0][i]=prefix[0][i-1]*nums[i-1];
            prefix[1][n-i-1]=prefix[1][n-i]*nums[n-i];
        }
        vector<int> arr;
        for(int i=0;i<n;i++)
        {
            arr.push_back(prefix[0][i]*prefix[1][i]);
        }
        return arr;
    }
};

File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.py
# Last updated: 2/21/2026, 9:44:04 AM
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n=len(nums)
        left_prod=[1]*n
        right_prod=[1]*n
        left_prod[0]=1

        for i in range(1,n,1):
                left_prod[i]=left_prod[i-1]*nums[i-1]

        right_prod[n-1]=1
        for i in range(n-2,-1,-1):
                right_prod[i]=right_prod[i+1]*nums[i+1]

        answer=[]
        for i in range(n):
            answer.append(right_prod[i]*left_prod[i])
        return answer